storage/api.go

```diff
@@ -53,16 +59,22 @@ func (b *QueryBatch) Queue(cmd string, args ...interface{}) {
 
 // Extend merges another batch into the current batch.
 func (b *QueryBatch) Extend(qb *QueryBatch) {
+	b.mu.Lock()
+	defer b.mu.Unlock()
```

Comment:
This should probably? also lock `qb` as it accesses `qb.items`.

Reply:
good eye; updated!